src/VaryingAttributes.cpp

```diff
@@ -601,7 +602,29 @@ class CastVariablesToFloatAndOffset : public IRMutator {
             }
         }
 
-        expr = Select::make(op->condition, mutated_true_value, mutated_false_value);
+        expr = Select::make(mutated_condition, mutated_true_value, mutated_false_value);
+    }
+
+    virtual void visit(const Ramp *op) {
+        Expr mutated_base = mutate(op->base);
+        Expr mutated_stride = mutate(op->stride);
+
+        // If either base or stride is a float, then make sure both are float
```

Comment:
We only want to cast in the case where this class has mutated an int to a float; any other type changes might introduce errors or mask other issues. Therefore, I'd rather not call match_types, which could make many other changes.